README.md

```diff
@@ -83,171 +104,205 @@ assert.deepStrictEqual(
+assert.deepStrictEqual(
+  collection.withResource({ id: 'a', name: 'Alice' }).find('a'),
+  RemoteData.success({ id: 'a', name: 'Alice' })
+);
 
-### Signature
-```ts
-withResourceAt(at: string, id: string, resource: Resource): RemoteCollection<Resource>
+assert.deepStrictEqual(
+  collection.withResourceFailure('b', 'Not authorized').find('b'),
+  RemoteData.failure(['Not authorized'])
+);
 ```
 
-## withListFailure
+## remove
 
-If requesting the list fails, calling this method will store a `RemoteFailure`
-with the passed `string`.
+Removes the resource indicated by the `id`.
 
 ### Signature
 ```ts
-withListFailure(error: string): Collection<Resource>
+remove(id: string): RemoteCollection<Resource>
 ```
 
-## withListFailureAt
-
-Like `withListFailure` but adds the failure only to the stored list of resources
-at the specified key, but doesn't interfere with other keys or the resource map
-itself.
+### Examples
 
-### Signature
 ```ts
-withListFailureAt(at: string, error: string): Collection<Resource>
+const users: User[] = [{ id: 'a', name: 'Alice' }, { id: 'b', name: 'Bob' }];
+const collection = new RemoteCollection<User>('id').withList(users);
+  
+assert.deepStrictEqual(
+  collection.view(),
+  RemoteData.success(users)
+);
+
+assert.deepStrictEqual(
+  collection.remove('a').view(),
+  RemoteData.success([users[1]])
+);
 ```
 
-## withResourceFailure
+# Collection Methods
 
-If requesting a single resource fails, calling this method will store a
-`RemoteFailure` with the passed `string` in the resource map for just this
-resource.
+## refresh
+
+Like `fetch` but for the whole list. The optional `at` arguments allows you to specify
+the view key to refresh.
 
 ### Signature
 ```ts
-withResourceFailure(id: string, error: string): RemoteCollection<Resource>
+refresh(at?: string = RemoteCollection.DEFAULT_KEY): RemoteCollection<Resource>
 ```
 
-## withResourceFailureAt
-
-The `At` version differs only in that requesting the view at the specified key
-will now see the newly errored resource.
-
-### Signature
+### Examples
 ```ts
-withResourceFailureAt(at: string, id: string, error: string): RemoteCollection<Resource>
+const users: User[] = [{ id: 'a', name: 'Alice' }, { id: 'b', name: 'Bob' }];
+const collection = new RemoteCollection<User>('id');
+
+assert.deepStrictEqual(
+  collection.refresh().view(),
+  RemoteData.pending
+);
+  
+assert.deepStrictEqual(
+  collection.withList(users).refresh().view(),
+  RemoteData.refresh(users)
+);
+  
+assert.deepStrictEqual(
+  collection.withList(users, 'team1').refresh('team1').view(),
+  RemoteData.refresh(users)
+);
 ```
 
-## find
+## withList
 
-Lookup a resource by its `idProp`. **NOTE**: If the resource has never been
-requested, or has been removed, we return `RemoteInitial` indicating that is has
-not been retrieved.
+Set or replace the list of resources.
+
+This method adds a list of resources to the `RemoteCollection` representing the
+normal success-case for fetching the collection.
```

Comment:
Can you add an additional description about the optional `at` argument?

Reply:
Check it now. I also did some other refactoring to eliminate always needing to pass the `idProp` or the actual `id`.